controllers/helmrepository_controller_oci.go

```diff
@@ -216,7 +208,28 @@ func (r *HelmRepositoryOCIReconciler) reconcileDelete(ctx context.Context, obj *
 func (r *HelmRepositoryOCIReconciler) notify(oldObj, newObj *sourcev1.HelmRepository, res sreconcile.Result, resErr error) {
 	// Notify successful recovery from any failure.
 	if resErr == nil && res == sreconcile.ResultSuccess {
-		if sreconcile.FailureRecovery(oldObj, newObj, helmRepositoryOCIFailConditions) {
+		// since the object is successfully reconciled without error,
+		// we can determine recovery by comparing the old and new object
+		// readyConditions status.
+		// We can use the readyConditions to determine if the object is
+		// recovered from a failure because in this reconciliation loop,
```

Comment:
I think this could be either 
```suggestion
		// recovered from a failure because, in this reconciliation loop,
```
or 
```suggestion
		// recovered from a failure because in this reconciliation loop
```